lib/elixir_analyzer/exercise_test/common_checks/private_helper_functions.ex

```diff
@@ -111,4 +117,79 @@ defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.PrivateHelperFunctions do
     for(_ <- 1..arity//1, do: "_")
     |> Enum.join(", ")
   end
+
+  @exceptional_enum_sort_functions %{
+    sort: %{arity: 2, sorter_argument_position: 2},
+    sort_by: %{arity: 3, sorter_argument_position: 3}
+  }
+
+  defp find_exceptions(
+         {:|>, _,
+          [
+            _,
+            {{:., [], [{:__aliases__, _, [:Enum]}, function_name]}, [], arguments}
+          ]} = node,
+         %{module: module, exceptions: exceptions}
+       ) do
+    exceptions =
+      if is_map(@exceptional_enum_sort_functions[function_name]) and
+           @exceptional_enum_sort_functions[function_name].arity == length(arguments) + 1 do
+        sorter_arg =
+          Enum.at(
+            arguments,
+            @exceptional_enum_sort_functions[function_name].sorter_argument_position - 2
+          )
+
+        find_enum_sort_exceptions(sorter_arg, %{module: module, exceptions: exceptions})
+      else
+        exceptions
+      end
+
+    {node, %{module: module, exceptions: exceptions}}
+  end
+
+  defp find_exceptions(
+         {{:., [], [{:__aliases__, _, [:Enum]}, function_name]}, [], arguments} = node,
+         %{module: module, exceptions: exceptions}
+       ) do
+    exceptions =
+      if is_map(@exceptional_enum_sort_functions[function_name]) and
+           @exceptional_enum_sort_functions[function_name].arity == length(arguments) do
+        sorter_arg =
+          Enum.at(
+            arguments,
+            @exceptional_enum_sort_functions[function_name].sorter_argument_position - 1
+          )
+
+        find_enum_sort_exceptions(sorter_arg, %{module: module, exceptions: exceptions})
+      else
+        exceptions
+      end
+
+    {node, %{module: module, exceptions: exceptions}}
+  end
+
+  defp find_exceptions(node, acc), do: {node, acc}
+
+  defp find_enum_sort_exceptions(sorter_arg, %{module: module, exceptions: exceptions}) do
+    case sorter_arg do
+      order when order in [:asc, :desc] ->
+        exceptions
+
+      {:__aliases__, _, module} when is_list(module) ->
+        [{[module], :def, :compare, 2} | exceptions]
+
+      {:__MODULE__, _, _} when is_list(module) ->
+        [{module, :def, :compare, 2} | exceptions]
```

Comment:
The use of `module` is a bit confusing in some of these cases, because this one refers to the `module` from the function argument, but in other cases, they are the module from `sorter_arg`. Those two are very close but not quite the same, so I would suggest a different name. Maybe `arg_module`?